src/transformers/quantizers/quantizer_bnb_4bit.py

```diff
@@ -207,11 +208,17 @@ def create_quantized_param(
                     if unexpected_keys is not None and k in unexpected_keys:
                         unexpected_keys.remove(k)
 
+            param_kwargs = {}
+            sig = inspect.signature(bnb.nn.Params4bit.from_prequantized)
+            if "module" in sig.parameters:
+                param_kwargs["module"] = module
+
```

Comment:
Quick question about `inspect.signature`, does this make the quantization slower ? I'm a bit concerned since  create_quantized_param is called for each param cc @muellerzr as I remember that you prefer not to use `inspect`

Reply:
@SunMarc also happy to do it has 

```
    @cached_property
    def is_bnb_gte_0_43_3(self) -> bool:
        if self._is_bnb_gte_0_43_3 is not None:
            return self._is_bnb_gte_0_43_3

        self._is_bnb_gte_0_43_3 = version.parse(importlib.metadata.version("bitsandbytes")) >= version.parse("0.43.3")

```
and then we can use `self. is_bnb_gte_0_43_3()`

Reply:
Version parsing is far more preferred!

Reply:
Though generally this would exist in `import_utils` as a `min_version` to `is_bnb_available()`. (We should then cache the bnb_version if we don't already, similar to `_torch_version`)

Reply:
Minor nit: I think if there's a variable or function named for this that the name should be more meaningful than `is_bnb_gte_0_43_3`. It should describe what that version check is about, e.g. 'bnb_can_load_from_float_storage`.

That said, I like the suggestion to model it the way I see `is_accelerate_available(min_version)` in `import_utils`.

Reply:
can we do the version parsing? 🤗